docs/user-guide/api-mediation/configuration-at-tls.md

```diff
@@ -1,23 +1,183 @@
+
+```pagent
+TTLSRule ApimlServerRule
+{
+  LocalAddr All
+  RemoteAddr All
+  LocalPortRange 10310-10320
+  Jobname ZWE*
+  Direction Inbound
+  TTLSGroupActionRef ServerGroupAction
+  TTLSEnvironmentActionRef ApimlServerEnvironmentAction
+  TTLSConnectionActionRef ApimlServerConnectionAction
+}
+
+TTLSGroupAction ServerGroupAction
+{
+  TTLSEnabled On
+}
+
+TTLSEnvironmentAction ApimlServerEnvironmentAction
+{
+  HandshakeRole ServerWithClientAuth
+  EnvironmentUserInstance 0
+  TTLSEnvironmentAdvancedParmsRef ServerEnvironmentAdvParms
+  TTLSKeyringParmsRef ApimlKeyring
+}
+
+TTLSConnectionAction ApimlServerConnectionAction
+{
+  HandshakeRole ServerWithClientAuth
+  TTLSCipherParmsRef CipherParms
+  TTLSConnectionAdvancedParmsRef ApimlConnectionAdvParms
+}
+```
+
+Note the setting `HandshakeRole` as this is meant for the core services which authenticate through certificates with each other and will allow the API Gateway to receive and accept X.509 client certificates from API Clients.
+
+### Outbound rules
+
+- For z/OSMF
+
+```pagent
+TTLSRule ApimlZosmfClientRule
+{
+  LocalAddr All
+  LocalPortRange 1024-65535
+  RemoteAddr All
+  RemotePortRange 449
+  Jobname ZWEAAG*
+  Direction Outbound
+  TTLSGroupActionRef ClientGroupAction
+  TTLSEnvironmentActionRef ApimlClientEnvironmentAction
+  TTLSConnectionActionRef ApimlNoX509ClientConnAction
+}
+
+TTLSGroupAction ClientGroupAction
+{
+  TTLSEnabled ON
+}
+
+TTLSEnvironmentAction ApimlClientEnvironmentAction
+{
+  HandshakeRole Client
+  TTLSKeyringParmsRef ApimlKeyring
+  TTLSCipherParmsRef CipherParms
+  TTLSEnvironmentAdvancedParmsRef ClientEnvironmentAdvParms
+}
+```
+
+Note `Jobname` is defined explicitly for the API Gateway, this is formed with the `zowe.job.prefix` setting from `zowe.yaml` plus `AG` as Gateway identifier.
+
+- For communication between API Gateway and other core services.
+
+```pagent
+TTLSRule ApimlClientRule
+{
+  LocalAddr All
+  LocalPortRange 1024-65535
+  RemoteAddr All
+  RemotePortRange 10310-10320
+  Jobname ZWEA*
+  Direction Outbound
+  TTLSGroupActionRef ClientGroupAction
+  TTLSEnvironmentActionRef ApimlClientEnvironmentAction
+  TTLSConnectionActionRef ApimlX509ClientConnAction
+}
+```
+
+- For communication between API Gateway and southbound services.
+
+```pagent
+TTLSRule ApimlServiceClientRule
+{
+  LocalAddr All
+  LocalPortRange 1024-65535
+  RemoteAddr All
+  RemotePortRange 40030
+  Jobname ZWEAAG*
+  Direction Outbound
+  TTLSGroupActionRef ClientGroupAction
+  TTLSEnvironmentActionRef ApimlClientEnvironmentAction
+  TTLSConnectionActionRef ApimlNoX509ClientConnAction
+}
+```
+
+- The outbound connection from the Gateway to the Discovery Service must not be configured with sending the server certificate.
+- Outbound connections from the Gateway to southbound services (onboarded services) must not send the server certificate if the service accepts x.509 Client Certificate authentication.
```

Comment:
As I mentioned above, the usage of AT-TLS is limited:
- if the service does not use X509 - rule without X509 could be generated
- otherwise the rule remove supporting of client cert
- for all services that requires X509 support, do not create any rule, the Java TLS will be used.